crypto/armor/armor.go

```diff
@@ -8,21 +8,34 @@ import (
 	"golang.org/x/crypto/openpgp/armor" //nolint: staticcheck
 )
 
-func EncodeArmor(blockType string, headers map[string]string, data []byte) string {
+// ErrEncode represents an error from calling [EncodeArmor].
+type ErrEncode struct {
+	Err error
+}
+
+func (e *ErrEncode) Error() string {
```

Comment:
Ah, just noticed this now after a second pass. This should be:

```suggestion
func (e ErrEncode) Error() string {
```

We don't need to mutate any fields once an error's been constructed. Doing this also makes error construction slightly less cumbersome and more ergonomic (e.g. `ErrEncode{Err: err}` as opposed to `&ErrEncode{Err: err}`).

Applies to all the error types here.

Reply:
But that incurs a copy right? But I suppose the cost is negligible. I will change that as well.

Reply:
Yeah it does, but it's negligible relative to the simplicity/readability/ergonomic benefits. If we had evidence that it was really hurting performance we could optimize specific cases, but there isn't any such evidence as of yet.